### PR TITLE
October 2017 SIP minutes draft

### PR DESCRIPTION
**Pull request by darjutak**
Dear Reviewer(s),

Please take a close look at the minutes.
My goal is to publish them ASAP so I can proceed with SIP "advertisement""
Olaf will add certain parts that I struggled with, see below.

Thank you in advance!
Darja

### CODE REVIEW

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+---
+
+# Minutes
+
+The following agenda was distributed to attendees:
+
+|Topic|Reviewers| Accepted/Rejected |
+| --- | --- | --- |
+| [SIP-34: Right-Associative By-Name Operators](http://docs.scala-lang.org/sips/right-associative-by-name-operators.html) | Adriaan Moors | Accepted |
+| [SIP-35: Opaque types](http://docs.scala-lang.org/sips/opaque-types.html) | Sébastien Doeraene | Pending |
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @adriaanm please give the feedback to the author

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+| [SIP-33: Match infix and prefix types to meet expression rules](http://docs.scala-lang.org/sips/make-types-behave-like-expressions.html)| Josh Suereth | Pending |
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @jsuereth please give the feedback to the author (see below what was discussed, if you didn't already take a look (o:

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+About 40 minutes were dedicated to finding a common ground for the  direction of the project as well as the technical details that should be addressed going forward.
+
+**Adriaan** immediately pointed out that by taking on the "hard stuff" to deal with first will give clarity to what could be supported for the next year. He suggests tackling the more ambitious features first, by experimenting in a current macro system, is a "cheap way" to find out the "unknown". The goal would be to find out how to implement it in the next macro system.
+Challenges that can be taken on:
+[(https://youtu.be/aIc-o1pcRhw?t=1673)]
+- splicing trees, ending up with owner chains that are correct
+- hygiene
+- better tools for macro authors, to experiment in a current macro system
+
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @adriaanm please take a look and let me know if I got all you wanted to come across, thanks!

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+**Eugene**
+https://youtu.be/aIc-o1pcRhw?t=1877
+Agrees both with **Adriaan** and **Olaf**, on one hand "prototyping" in the current macro system can be beneficial e.g automatic owner chain fixer but on the other, scala reflect is fundamentally different from the prototype macro system and even if the tests are run there might be no point to it.
+He does agree with **Olaf** that classification of the macros is useful. Concluding that supporting the generation transformation macros shouldn't be that hard, but than the question of how valuable to the community these are needs to be raised.
+-- OLAF NEEDS TO RE-WRITE THIS PART --
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @olafurpg

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+ Go to: https://youtu.be/aIc-o1pcRhw?t=1960
+ "How restrictive is to demand all quasiquote to be fully typed instead of free name reference to type up the abstract" **Martin**
+-- PLEASE SUMMARISE THE MAIN POINTS IN THIS DISCUSSION --
+
+ - White box / Black box(by **Martin** / **Olaf**)
+  Main concern "when do tools have to run the untrusted code?"
+
+ - Annotation macros (Community)
+ **Martin** insists on leaving the paradise functionality in "meta/template scala" but that *scala* assumes only the things that have proven themselves e.g. simulacrum
+ -- PLEASE ADD THE REASON IF NECESSARY
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @olafurpg

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+ He strongly believes that macro annotations are a "complete abuse of macros" and will make sure that "this things are not possible anymore"
+
+- The interaction between type inference and macros (**Iulian**)
+quote https://youtu.be/aIc-o1pcRhw?t=3436
+-- OLAF, PLEASE MAKE SURE THE SENSE IS MADE OF WHAT IULIAN SAID --
+
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @olafurpg

File: _sips/minutes/2017-10-24-sip-minutes.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+**Adriaan** thinks that's a great example for the language feature.
+
+- Inline (**Martin**)
+"Keep inline as a sort of enabler of meta to get full macros. Main job moving the code from A to B, doing simplifications, wheres actual generation and inspection of quasiquote is the job of macros."
+Adding that it is also an interesting idea to split generation and inspection
+
+- Untype trees ??
+https://youtu.be/aIc-o1pcRhw?t=3752
+-- OLAF PLEASE PUT A SENTENCE OR TWO --
```
Comment (darjutak) [inline _sips/minutes/2017-10-24-sip-minutes.md]: @olafurpg

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 3 above)
Comment (adriaanm) [reply _sips/minutes/2017-10-24-sip-minutes.md]: Thanks, reads better than I said it :-)

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 5 above)
Comment (olafurpg) [reply _sips/minutes/2017-10-24-sip-minutes.md]: > **Martin** insists on leaving the paradise functionality in "paradise scala" and annotations that prove themselves to be essential can be considered for inclusion in the main scala.

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 6 above)
Comment (olafurpg) [reply _sips/minutes/2017-10-24-sip-minutes.md]: > **Iulian** would like to see addressed the interaction between type inference and macros. This interaction seems to be one of the hard unsolved problems in the current macro system. For example, code that uses shapeless typically has two type parameters where one parameter is unbounded and it's inferred type is guided by the shapeless macro.

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 7 above)
Comment (olafurpg) [reply _sips/minutes/2017-10-24-sip-minutes.md]: > **Eugene** thinks that mixing of untyped trees and typed trees is one of the hard problems that he believes we only a "few steps" away from solving. He believes solving this problem should be the focus of next month and the fruit of that work will benefit both the new macro system as well as existing scala.reflect macros and open opportunities for more macro applications. **Adriaan** comments that if we focus on the easy stuff and postpone the hard problems then there is a chance that we will never get to working on the hard problems. Getting some of the hard stuff out of the way now will give us clarity on what we can support in the next year.

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 3 above)
Comment (darjutak) [reply _sips/minutes/2017-10-24-sip-minutes.md]: @adriaanm ohhh, you are being too nice (o: I re-wrote a bit, so it's even better hihihi

File: _sips/minutes/2017-10-24-sip-minutes.md (hunk 6 above)
Comment (darjutak) [reply _sips/minutes/2017-10-24-sip-minutes.md]: @olafurpg THANK YOU! (this goes for all the comments below)

**Pull request merged**